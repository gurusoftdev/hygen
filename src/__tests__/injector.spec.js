const injector = require('../ops/injector')
const gemfile = `
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    `
describe('injector', () => {
  it('before rails', () => {
    expect(
      injector(
        {
          attributes: {
            before: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('after rails', () => {
    expect(
      injector(
        {
          attributes: {
            after: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('prepend top of file', () => {
    expect(
      injector(
        {
          attributes: {
            prepend: true
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('append bottom of file', () => {
    expect(
      injector(
        {
          attributes: {
            append: true
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('at_index 2 (below "source")', () => {
    expect(
      injector(
        {
          attributes: {
            at_line: 2
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('skip_if "source" exists', () => {
    expect(
      injector(
        {
          attributes: {
            skip_if: 'source',
            after: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
})
