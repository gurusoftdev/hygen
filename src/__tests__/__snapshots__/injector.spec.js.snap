// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injector after rails 1`] = `
"
    source 'http://rubygems.org'
    gem 'rails'
    gem 'kamikaze' # added by hygen
    gem 'nokogiri'
    gem 'httparty'

    "
`;

exports[`injector append bottom of file 1`] = `
"
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    gem 'kamikaze' # added by hygen
    "
`;

exports[`injector at_index 2 (below "source") 1`] = `
"
    source 'http://rubygems.org'
    gem 'kamikaze' # added by hygen
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    "
`;

exports[`injector before rails 1`] = `
"
    source 'http://rubygems.org'
    gem 'kamikaze' # added by hygen
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    "
`;

exports[`injector prepend top of file 1`] = `
"    gem 'kamikaze' # added by hygen

    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    "
`;

exports[`injector skip_if "source" exists 1`] = `
"
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    "
`;
